bin/rucio-admin

```diff
@@ -1003,18 +1004,57 @@ def declare_bad_file_replicas(args):
         for pfn in bad_files_pfns:
             print(pfn)
 
-    chunk_size = 10000
-    tot_files = len(bad_files)
-    cnt = 0
-    nchunk = math.ceil(tot_files / chunk_size)
-    for chunk in chunks(bad_files_pfns, chunk_size):
-        cnt += 1
-        client.add_bad_pfns(pfns=chunk, reason=args.reason, state='BAD', expires_at=None)
-        ndeclared = len(chunk)
-        print('Chunk %s/%s : %s replicas successfully declared' % (int(cnt), int(nchunk), ndeclared))
-    print('--------------------------------')
-    print('Summary')
-    print('%s replicas successfully declared' % tot_files)
+    if len(bad_files_pfns) < 100:
+        # Using the old API to declare
+        non_declared = client.declare_bad_file_replicas(pfns=bad_files_pfns, reason=args.reason)
+        for rse in non_declared:
+            for pfn in non_declared[rse]:
+                print('%s : PFN %s cannot be declared.' % (rse, pfn))
+    else:
+        print('Getting the information about RSE protocols. It can take several seconds')
+        dict_rse = client.export_data(distance=False)
+        prot_dict = {}
+        for rse, dict_attr in dict_rse['rses'].items():
+            protocols = dict_attr['protocols']
+            for prot in protocols:
+                if prot['scheme'] == 'srm':
+                    prot_dict[str('%s://%s%s' % (prot['scheme'], prot['hostname'], prot['prefix']))] = rse
+                else:
+                    prot_dict[str('%s://%s:%s%s' % (prot['scheme'], prot['hostname'], prot['port'], prot['prefix']))] = rse
```

Comment:
Sorry, I didn’t try to run the new code, but my understanding is that this would require explicitly setting the port in the PFNs. Is that the case?

Reply:
Constraint relaxed. Both are accepted